scico/ladmm.py

```diff
@@ -0,0 +1,365 @@
+# -*- coding: utf-8 -*-
```

Comment:
With the addition of these optimizers, I find having each optimizer as a top-level module odd. I.e., looking at the table of contents for the API reference (https://scico--117.org.readthedocs.build/en/117/_autosummary/scico.html), we have `scico.admm` and `scico.ladmm` at the same level as `scico.linop`. I propose that it's time for a `scico.optimizer` package.

Reply:
That's a good suggestion, but there are a few questions to resolve regarding the implementation:

- Naming:
  - Instead of `optimizer`, I would go with `optimize`, for consistency with `scipy.optimize`.
  - Are we going to also move `scico.solver` into this subpackage? If not, the name should perhaps reflect specific properties that differentiate it from the functions in `scico.solver`, e.g. `scico.proxopt`, since all the algorithms that would be within the subpackage would be proximal algorithms.
- Structure. Do we
  - retain the existing modules, so that e.g. `scico.admm.ADMM` becomes ` `scico.optimize.admm.ADMM`, or
  - follow the same approach as in e.g.`scico.linop`, so that `scico.admm.ADMM` becomes `scico.optimize.ADMM`.
  
My leaning is towards naming it `scico.optimize`, which would include `scico.solver`, and making the classes direct members of this module, e.g. `scico.optimize.ADMM`.

Reply:
Discussion moved to #122.